### ISSUE TITLE
`pnpm` fails to properly create `expo-app` project (but `yarn` succeeds)

### ISSUE THREAD

**Issue by Andrew15-5** [pnpm/pnpm#6597]
### pnpm version:

8.5.1

### Code to reproduce the issue:

```sh
cd /tmp
mkdir projects
cd projects
\yarn create expo-app yarn-app && cd yarn-app/ && \yarn start # works just fine
# <Ctrl+C>
cd ..
\pnpm create expo-app pnpm-app && cd pnpm-app/ && \pnpm start # doesn't work
```

logs of `yarn start`:

<details><summary>yarn start logs</summary>

```text
yarn run v1.22.19
$ expo start
Starting project at /tmp/projects/yarn-app
Starting Metro Bundler
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █   █▄ ▀▄▀█ ▄▄▄▄▄ █
█ █   █ █ ▀▄ █▀█ ▄█ █   █ █
█ █▄▄▄█ █▀██▀▀ █▄██ █▄▄▄█ █
█▄▄▄▄▄▄▄█▄▀▄█ █▄█▄█▄▄▄▄▄▄▄█
█▄▄█▀▄▄▄▀█▀▀▄▀█▄ ███ ▀▄▄ ▄█
█▀█▄ ▄ ▄▄██▀ ▄▄█  ▀ █▄  ▀██
█▀▄ █▄▄▄ ▀▀▄█▄▀▄▀▄▀▄▀▀▄ ▀██
███▀▄▀▄▄▄▀▄▄█▀█▀▄▄▄█▄▀ ▀███
█▄▄▄▄█▄▄█  ▄▄ █▄▄ ▄▄▄ ▀ ▄▄█
█ ▄▄▄▄▄ █▀ ▄ ██▀▀ █▄█ ▀▀███
█ █   █ █▄█ ▀▀▀▄█▄▄ ▄▄▀ ▀▀█
█ █▄▄▄█ █▀▀▀ ███▄██▄▀█▀▀ ██
█▄▄▄▄▄▄▄█▄▄▄███▄████▄▄▄▄▄▄█

› Metro waiting on exp://192.168.0.104:19000
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
› Reloading apps
Android Bundling complete 2467ms
› Stopped server
Done in 14.64s.
```

</details>

Logs of `pnpm start`:

<details><summary>pnpm start logs</summary>

```text
> pnpm-app@1.0.0 start /tmp/projects/pnpm-app
> expo start

Starting project at /tmp/projects/pnpm-app
Starting Metro Bundler
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █   █▄ ▀▄▀█ ▄▄▄▄▄ █
█ █   █ █ ▀▄ █▀█ ▄█ █   █ █
█ █▄▄▄█ █▀██▀▀ █▄██ █▄▄▄█ █
█▄▄▄▄▄▄▄█▄▀▄█ █▄█▄█▄▄▄▄▄▄▄█
█▄▄█▀▄▄▄▀█▀▀▄▀█▄ ███ ▀▄▄ ▄█
█▀█▄ ▄ ▄▄██▀ ▄▄█  ▀ █▄  ▀██
█▀▄ █▄▄▄ ▀▀▄█▄▀▄▀▄▀▄▀▀▄ ▀██
███▀▄▀▄▄▄▀▄▄█▀█▀▄▄▄█▄▀ ▀███
█▄▄▄▄█▄▄█  ▄▄ █▄▄ ▄▄▄ ▀ ▄▄█
█ ▄▄▄▄▄ █▀ ▄ ██▀▀ █▄█ ▀▀███
█ █   █ █▄█ ▀▀▀▄█▄▄ ▄▄▀ ▀▀█
█ █▄▄▄█ █▀▀▀ ███▄██▄▀█▀▀ ██
█▄▄▄▄▄▄▄█▄▄▄███▄████▄▄▄▄▄▄█

› Metro waiting on exp://192.168.0.104:19000
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Press a │ open Android
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
› Reloading apps
warn No apps connected. Sending "reload" to all React Native apps failed. Make sure your app is runn
ing in the simulator or on a phone connected via USB.
› Reloading apps
warn No apps connected. Sending "reload" to all React Native apps failed. Make sure your app is runn
ing in the simulator or on a phone connected via USB.
› Reloading apps
Error: Unable to resolve module ./node_modules/expo/AppEntry from /tmp/projects/pnpm-app/.:

None of these files exist:
  * node_modules/expo/AppEntry(.native|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx|.and
roid.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json)
  * node_modules/expo/AppEntry/index(.native|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.ts
x|.android.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json)
    at ModuleResolver.resolveDependency (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node
_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:137:15)
    at DependencyGraph.resolveDependency (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/nod
e_modules/metro/src/node-haste/DependencyGraph.js:260:43)
    at /tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node_modules/metro/src/lib/transformHe
lpers.js:177:21
    at Server._resolveRelativePath (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node_modu
les/metro/src/Server.js:1161:12)
    at async Server.requestProcessor [as _processBundleRequest] (/tmp/projects/pnpm-app/node_modules
/.pnpm/metro@0.73.9/node_modules/metro/src/Server.js:468:37)
    at async Server._processRequest (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node_mod
ules/metro/src/Server.js:390:9)
› Stopped server
```

</details>

Even with the simple template (I was trying with `-t expo-template-blank-typescript` at first) pnpm fails to deliver.

### Expected behavior:

`pnpm-app` should run with no errors on the remote device like `yarn-app` did.

### Actual behavior:

pnpm gives an error:

```text
Error: Unable to resolve module ./node_modules/expo/AppEntry from /tmp/projects/pnpm-app/.:

None of these files exist:
  * node_modules/expo/AppEntry(.native|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx|.and
roid.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json)
  * node_modules/expo/AppEntry/index(.native|.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.ts
x|.android.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json)
    at ModuleResolver.resolveDependency (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node
_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:137:15)
    at DependencyGraph.resolveDependency (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/nod
e_modules/metro/src/node-haste/DependencyGraph.js:260:43)
    at /tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node_modules/metro/src/lib/transformHe
lpers.js:177:21
    at Server._resolveRelativePath (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node_modu
les/metro/src/Server.js:1161:12)
    at async Server.requestProcessor [as _processBundleRequest] (/tmp/projects/pnpm-app/node_modules
/.pnpm/metro@0.73.9/node_modules/metro/src/Server.js:468:37)
    at async Server._processRequest (/tmp/projects/pnpm-app/node_modules/.pnpm/metro@0.73.9/node_mod
ules/metro/src/Server.js:390:9)
```

### Additional information:

- `node -v` prints: v18.16.0
- Windows, macOS, or Linux?: Pop!_OS 22.04
- Remote device for testing the apps: Android 12 (physical)

**Comment by baryon** [comment]
another issue: https://github.com/expo/expo/issues/22580
please check it too.

> we had to revert that upgrade because more recent versions of uuid have native dependencies in the react-native environment. if you are seeing this error, there is a problem with how pnpm is constructing your node_modules tree, such that the incorrect version of `uuid` is imported by the `expo` package. `expo` depends on `uuid@^3.4.0` and if another version is being imported that is a problem with the package manager

**Comment by Andrew15-5** [comment]
I'm sorry, I really don't understand what is (specifically) being discussed in the mentioned issue, as I am just a beginner with React Native and expo app. And therefore I don't really know what I need to check. And I'm not that knowledgeable with all the dependencies problems that are present in NPM packages.

P.S. I'm not actually interested in the issue (anymore; as of right now). I used the `yarn dlx react-native init` (yarn berry) command (it uses TypeScript by default) to start my project after all the testing and playing with different ways to create/init a React Native app.

**Comment by Andrew15-5** [comment]
If you need me to test anything, just tell the command (or the steps) and I'll give feedback.

**Comment by zkochan** [comment]
Closing for lack of a reproduction on a supported version.

The reproduction is three years old and the template has changed several times since. Please open a new issue with a reproduction if creating the project still fails on a current pnpm.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
